File: lesson_8/app/src/main/java/com/kalashnyk/denys/gmapsapp/presentation/activities/splash/SplashActivity.java
package com.kalashnyk.denys.gmapsapp.presentation.activities.splash;

import android.os.Handler;
import android.os.Bundle;

import com.kalashnyk.denys.gmapsapp.R;
import com.kalashnyk.denys.gmapsapp.di.component.ViewModelComponent;
import com.kalashnyk.denys.gmapsapp.presentation.activities.main.MainActivity;
import com.kalashnyk.denys.gmapsapp.presentation.base.BaseActivity;

public class SplashActivity extends BaseActivity {

    private long SPLASH_TIMER = 3000L;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        new Handler().postDelayed(() -> {
            startActivity(MainActivity.newInstance(this));
            finish();
        }, SPLASH_TIMER);
    }

    @Override
    protected void injectDependency(ViewModelComponent component) { }
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/presentation/activities/splash/SplashActivity.kt
package com.kalashnyk.denys.kotlinbindingsample.presentation.activities.splash

import android.databinding.DataBindingUtil
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import com.kalashnyk.denys.kotlinbindingsample.R
import com.kalashnyk.denys.kotlinbindingsample.databinding.ActivitySplashBinding
import com.kalashnyk.denys.kotlinbindingsample.presentation.activities.main.MainActivity

class SplashActivity : AppCompatActivity() {

    private val timer = 3000L
    private lateinit var binding: ActivitySplashBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_splash)
        Handler().postDelayed({
            startActivity(MainActivity.newInstance(this))
            finish()
        }, timer)
    }
}
